### ISSUE TITLE
Crash with accessibility next/prev events and min/max date

### ISSUE THREAD

**Issue by laalto** [wdullaer/MaterialDateTimePicker#549]
Library version 4.0.1.

Steps to reproduce:

1. Enable TalkBack (or some other accessibility helper with next/previous navigation)
2. Open up a DatePickerDialog with a minimum date set e.g. to current date
3. Select first day of current month
4. Swipe left to move to previous focusable item

Expected:

Invalid dates (before minimum date) should not be focused?
No crash, focus moves e.g. to controls above the month selection view.

Actual:

Invalid dates can be focused with accessibility navigation.
Crash trying to move to previous month that is before minimum date.

```
java.lang.IllegalArgumentException: Invalid target position
	at androidx.recyclerview.widget.RecyclerView$SmoothScroller.start(RecyclerView.java:11534)
	at androidx.recyclerview.widget.RecyclerView$LayoutManager.startSmoothScroll(RecyclerView.java:8204)
	at androidx.recyclerview.widget.LinearLayoutManager.smoothScrollToPosition$6cefd9a5(LinearLayoutManager.java:451)
	at androidx.recyclerview.widget.RecyclerView.smoothScrollToPosition(RecyclerView.java:1781)
	at com.wdullaer.materialdatetimepicker.date.DayPickerView.goTo$4001730e(DayPickerView.java:228)
	at com.wdullaer.materialdatetimepicker.date.DayPickerView.performAccessibilityAction(DayPickerView.java:422)
	at android.view.AccessibilityInteractionController.performAccessibilityActionUiThread(AccessibilityInteractionController.java:711)
```

Assuming this crashes the same way with maximum date set respectively; did not try to repro that myself.

This looks a lot like #512 reported earlier.

**Comment by wdullaer** [comment]
I don't do a lot of testing with the accessibility features.

While it looks similar to #512 it's probably not the same issue.

I'll take a closer look at this.

**Comment by laalto** [comment]
I did some experimentation with this and submitted a PR that fixes the issue at least for our case. Not too familiar with the codebase so might have overlooked something but hope it's helpful anyway.

**Comment by wdullaer** [comment]
Fixed in v4.1.0
